frontend/pkg/frontend/helpers.go

```diff
@@ -17,6 +18,57 @@ import (
 	"github.com/Azure/ARO-HCP/internal/database"
 )
 
+// CheckForConflict returns a "409 Conflict" error response if the provisioning
+// state of the resource is non-terminal, or any of its parent resources within
+// the same provider namespace are in a "Deleting" state.
+func (f *Frontend) CheckForConflict(ctx context.Context, operationRequest database.OperationRequest, doc *database.ResourceDocument) *arm.CloudError {
+	switch operationRequest {
+	case database.OperationRequestCreate:
```

Comment:
note to future reviewers: this is a [golang switch statement no-op case](https://go.dev/wiki/Switch#noop-case) not something like other languages where `Create` and `Delete` have the same path because there is no body defined; Figured I'd save some searches

Reply:
Yes.  In Go, case fallthroughs are explicit.  Implicit fallthroughs are a source of many bugs in C/C++ code.